plugins/ingest-attachment/src/main/java/org/elasticsearch/ingest/attachment/AttachmentProcessor.java

```diff
@@ -197,6 +203,7 @@ int getIndexedChars() {
         public AttachmentProcessor create(Map<String, Processor.Factory> registry, String processorTag,
                                           String description, Map<String, Object> config) throws Exception {
             String field = readStringProperty(TYPE, processorTag, config, "field");
+            String fileName = readStringProperty(TYPE, processorTag, config, "file_name", "file_name");
```

Comment:
Also, is there any particular reason why you chose `file_name` as the property name when it's called `resourceName` on the Tika side? Unless there's a good reason for `file_name`, it's probably best to stick with `resource_name` since that's what Tika calls the property.